pkg/operator/hive/hive.go

```diff
@@ -42,6 +43,13 @@ const (
 	// hiveConfigHashAnnotation is annotation on hivedeployment that contains
 	// the hash of the contents of the hive-controllers-config configmap
 	hiveConfigHashAnnotation = "hive.openshift.io/hiveconfig-hash"
+
+	// hiveImagePullSecretLabel is the key (the value will always be "true") for the label we add to
+	// the secret (named in HiveConfig.Spec.HiveImagePullSecretRef) that is copied into the target
+	// hive namespace (named in HiveConfig.Spec.TargetNamespace). We use this to identify secrets in
+	// namespaces that were *previously* the configured TargetNamespace so we can clean
+	// them up.
```

Comment:
Why are we not just using owner references? I guess they would have to be on one of the Deployments/StatefulSets, but that should work, shouldn't it?